packages/itwinui-react/src/core/Overlay/Overlay.tsx

```diff
@@ -0,0 +1,101 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import {
+  polymorphic,
+  type PolymorphicForwardRefComponent,
+} from '../utils/index.js';
+import cx from 'classnames';
+
+export type OverlayComponentProps = {
+  /**
+   * The sub-component elements of Overlay.
+   */
+  children?: React.ReactNode;
+  /**
+   * Placeholder for Progress Indicator in Overlay
+   */
+  content?: React.ReactNode;
+};
+
+// --------------------------------------------------------------------------------
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const OverlayComponent = React.forwardRef((props, forwardedRef) => {
+  const { content, children, ...rest } = props;
+
+  return (
+    <OverlayWrapper {...rest}>
+      <OverlayMessage>{content}</OverlayMessage>
+      <OverlayHiddenContent>{children}</OverlayHiddenContent>
+    </OverlayWrapper>
+  );
+}) as PolymorphicForwardRefComponent<'div', OverlayComponentProps>;
+OverlayComponent.displayName = 'Overlay';
+
+// --------------------------------------------------------------------------------
+
+const OverlayHiddenContent = React.forwardRef((props, ref) => {
+  const { as: Box = 'div', children, ...rest } = props;
+  return (
+    <Box custom-attribute='inert' ref={ref} {...rest}>
```

Comment:
this custom attribute has no effect. the resulting html needs to be `<div inert>`, not `<div custom-attribute='inert'>`

see docs: https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/inert

Reply:
react doesn't fully support `inert` for some reason, so you'll need to use this hack:

```suggestion
    <Box {...{ inert: '' }} ref={ref} {...rest}>
```